crates/apub/src/http/mod.rs

```diff
@@ -102,3 +106,13 @@ pub(crate) async fn get_activity(
     create_apub_response(&activity.data)
   }
 }
+
+pub fn check_community_valid(community: &Community) -> LemmyResult<()> {
+  if community.deleted || community.removed {
```

Comment:
This check appears duplicated in some of the above code. Trace down every call of `check_community_valid` to make sure you are only checking things once.

Reply:
Also maybe this should be renamed to something like: `check_public_community_valid`, or `ensure_community_public` . 

LocalOnly communities could still be *valid*, but they'll fail this check.

Reply:
The checks in `get_apub_community_http()` etc are necessary to return the tombstone. The check for remove/delete is useful for methods like `get_apub_community_followers()` which can simply return an error. So I dont see a good way to change the code, but added a comment to the check function.